src/commands.rs

```diff
@@ -2062,18 +2124,7 @@ fn cmd_abandon(
     args: &AbandonArgs,
 ) -> Result<(), CommandError> {
     let mut workspace_command = command.workspace_helper(ui)?;
-    let to_abandon = {
-        let mut acc = Vec::new();
-        for revset in &args.revisions {
-            let revisions = workspace_command.resolve_revset(revset)?;
-            workspace_command.check_non_empty(&revisions)?;
-            for commit in &revisions {
-                workspace_command.check_rewriteable(commit)?;
-            }
-            acc.extend(revisions);
-        }
-        acc
-    };
+    let to_abandon = resolve_multiple_rewriteable_revsets(&args.revisions, &workspace_command)?;
```

Comment:
Not for this PR, but it seems that this would also need to be de-duplicated, so `jj abandon @ @` doesn't print the commit twice

Reply:
Now that we have `IndexSet`, it's trivial to do and won't change the order of things.

Reply:
I added this in. As I hoped, all I needed to do was to replace a `Vec` with an `IndexSet`. This meant that I also moved the commit with the other test for `jj abandon` towards the end of the PR.